src/main.rs

```diff
@@ -27,37 +33,50 @@ async fn main() -> Result<(), ()> {
         .0;
 
     let lnd_args = LndCfg::new(config.address, config.cert, config.macaroon);
-    let log_level = match config.log_level {
-        Some(level_str) => match LevelFilter::from_str(&level_str) {
-            Ok(level) => level,
-            Err(_) => {
-                // Since the logger isn't set up yet, we use a println just this once.
-                println!(
-                    "User provided log level '{}' is invalid. Make sure it is set to either 'error',
-                    'warn', 'info', 'debug' or 'trace'",
-                    level_str
-                );
-                return Err(());
-            }
-        },
-        None => LevelFilter::Info,
-    };
     let (shutdown, listener) = triggered::trigger();
-    // Create the channel which will tell us when the onion messenger has finished starting up.
-    let (tx, _): (Sender<u32>, Receiver<u32>) = mpsc::channel(1);
-    let signals = LifecycleSignals {
-        shutdown,
-        listener,
-        started: tx,
-    };
+    let signals = LifecycleSignals { shutdown, listener };
     let args = Cfg {
         lnd: lnd_args,
-        log_dir: config.log_dir,
-        log_level,
         signals,
     };
 
-    let handler = OfferHandler::new();
-    let messenger = LndkOnionMessenger::new(handler);
-    messenger.run(args).await
+    let handler = Arc::new(OfferHandler::new());
+    let messenger = LndkOnionMessenger::new();
+    setup_logger(config.log_level, config.log_dir)?;
+
+    let mut client = get_lnd_client(args.lnd.clone()).expect("failed to connect to lnd");
+    let info = client
+        .lightning()
+        .get_info(GetInfoRequest {})
+        .await
+        .expect("failed to get info")
+        .into_inner();
+
+    let grpc_port = match config.grpc_port {
+        Some(port) => port,
+        None => DEFAULT_SERVER_PORT,
+    };
+    let addr = format!("[::1]:{grpc_port}").parse().map_err(|e| {
```

Comment:
This only works if the lndk daemon is running on the same host as lnd. I'm trying to run this in a docker environment where lndk is running in it's own container.

The issue is that your gRPC server is binding to the address `[::1]`, which is the IPv6 equivalent of `127.0.0.1` or `localhost`. This means that the server is only listening for connections from within the same network (in this case, the Docker container).

To allow connections from outside the Docker container (such as from your host machine), you need to bind the server to the address `0.0.0.0` or `[::]` (the IPv6 equivalent). This will allow the server to listen for connections from any network.

Probably the listen address for the gRPC server needs to be configurable.

Reply:
Here is a commit which adds configurable grpc server host, which gets this working for my use case
https://github.com/lndk-org/lndk/commit/9f8d172c371df151ba89555edff5222b415dc1dd

Reply:
Good call, I forgot to configure that. I'll add your commit here.